extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.vault;
+
+import java.util.List;
+
+import io.quarkus.vault.transit.ClearData;
+import io.quarkus.vault.transit.DecryptionRequest;
+import io.quarkus.vault.transit.DecryptionResult;
+import io.quarkus.vault.transit.EncryptedData;
+import io.quarkus.vault.transit.EncryptionRequest;
+import io.quarkus.vault.transit.EncryptionResult;
+import io.quarkus.vault.transit.RewrappingRequest;
+import io.quarkus.vault.transit.Signature;
+import io.quarkus.vault.transit.SigningBatchRequest;
+import io.quarkus.vault.transit.SigningResult;
+import io.quarkus.vault.transit.TransitContext;
+import io.quarkus.vault.transit.VerificationRequest;
+import io.quarkus.vault.transit.VerificationResult;
+import io.quarkus.vault.transit.Verifyable;
+
+public interface VaultTransitSecretEngine {
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, clearData.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decrypt(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the string to encrypt
+     * @return cipher text
+     */
+    String encrypt(String keyName, String clearData);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the data to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    EncryptedData encrypt(String keyName, ClearData clearData, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of results, each being either a ciphertext or an error.
+     * Each element can specify an optional context, allowing key derivation.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return a list of result objects containing either a ciphertext or an error
+     */
+    List<EncryptionResult> encrypt(String keyName, List<EncryptionRequest> requests);
```

Comment:
IMHO it should return a List of Strings only and the method should throw `EncryptionException` if at least a single error is returned. I believe this version of `encrypt` is only about the optimizations (encrypt several pieces of the request or response related data in one go), I don't think the user code will continue operating if it gets some data encrypted, some - not - because all of those data are meant to be encrypted for a reason.
IMHO the same reasoning should be applied to all other `Result` beans

Reply:
I disagree here. for large batches I have seen people work with error threshold. they will fail the batch if they have 10% errors for instance, but will process as much as possible below that theshold (and create an error report with issues to be manually processed later)

Reply:
@vsevel Sure some power users may want to do that: IMHO though if Vault has failed to have some sensitive data encrypted, what is our recommendation to the Quarkus users - 'proceed if you think it makes sense for you' ? But that would send a message that may be we are not sure what that situation can really be about :-). I'm certainly not sure :-)
But even if let these cases happen, we can still have an exception which would report an error in a Java way, and let users access the good and bad results indirectly. Otherwise, for the typical case, one have to write:
```
EncryptionResult result = vault.encrypt(...);
if (result.getError() != null) {
   throw new VaultEncryptionError(result.getError());
} else {
   return result.getData();
}
```

Reply:
@sberyozkin I think I found a nice compromise (may be that is what you were hinting at?). I changed the api to:
```
List<String> encrypt(String keyName, List<EncryptionRequest> requests);
```
If there are any errors, you get a `VaultEncryptionBatchException` with a message saying `10 encryption errors` and a `List<EncryptionResult> results`, which provides all the details if you are a power user.

Reply:
@vsevel Yes, this is just perfect IMHO :-), thanks for going for a compromise